java/java-impl/src/com/intellij/codeInsight/FunctionalInterfaceSuggester.java

```diff
@@ -276,6 +269,20 @@ else if (expression instanceof PsiMethodReferenceExpression) {
                 }
               }
             }
+            else if (maybeDefaultConstructor(referenceExpression, parameters, element)) {
```

Comment:
This code is meant to show the idea: since `MethodReferenceResolver#resolve` encapsulates the default constructor into [`com.intellij.psi.infos.ClassCandidateInfo`](https://github.com/neshkeev/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/tree/java/MethodReferenceResolver.java#L65) it cannot be handled uniformly, so this if-else-branch fixes the problem, but I require guidance from the **jetbrains** team to make it beter